### ISSUE TITLE
[BUG] Connection to the server `api.crc.testing:6443` was refused

### ISSUE THREAD

**Issue by ruivieira** [code-ready/crc#1201]
### General information

  * OS: Linux
  * Hypervisor: KVM
  * (Did you run `crc setup` before starting it?) Yes
  * Contents of `lscpu`:

```
Architecture:                    x86_64
CPU op-mode(s):                  32-bit, 64-bit
Byte Order:                      Little Endian
Address sizes:                   43 bits physical, 48 bits virtual
CPU(s):                          24
On-line CPU(s) list:             0-23
Thread(s) per core:              2
Core(s) per socket:              12
Socket(s):                       1
NUMA node(s):                    1
Vendor ID:                       AuthenticAMD
CPU family:                      23
Model:                           113
Model name:                      AMD Ryzen 9 3900 12-Core Processor
Stepping:                        0
Frequency boost:                 enabled
CPU MHz:                         2195.414
CPU max MHz:                     3100.0000
CPU min MHz:                     2200.0000
BogoMIPS:                        6188.43
Virtualization:                  AMD-V
L1d cache:                       384 KiB
L1i cache:                       384 KiB
L2 cache:                        6 MiB
L3 cache:                        64 MiB
NUMA node0 CPU(s):               0-23
Vulnerability Itlb multihit:     Not affected
Vulnerability L1tf:              Not affected
Vulnerability Mds:               Not affected
Vulnerability Meltdown:          Not affected
Vulnerability Spec store bypass: Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:        Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:        Mitigation; Full AMD retpoline, IBPB conditional, STIBP conditional, RSB filling
Vulnerability Tsx async abort:   Not affected
Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb r
                                 dtscp lm constant_tsc rep_good nopl nonstop_tsc cpuid extd_apicid aperfmperf pni pclmulqdq monitor ssse3 fma cx16 sse4_1 sse4_2 movbe pop
                                 cnt aes xsave avx f16c rdrand lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs skinit wdt tce topoe
                                 xt perfctr_core perfctr_nb bpext perfctr_llc mwaitx cpb cat_l3 cdp_l3 hw_pstate sme ssbd mba sev ibpb stibp vmmcall fsgsbase bmi1 avx2 sm
                                 ep bmi2 cqm rdt_a rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local
                                  clzero irperf xsaveerptr rdpru wbnoinvd arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfth
                                 reshold avic v_vmsave_vmload vgif umip rdpid overflow_recov succor smca
```
   * Contents of `cat /proc/meminfo`:

```
MemTotal:       65868644 kB
MemFree:        16542240 kB
MemAvailable:   52207912 kB
Buffers:         1190088 kB
Cached:         33966544 kB
SwapCached:            0 kB
Active:         19152076 kB
Inactive:       27772216 kB
Active(anon):   11770700 kB
Inactive(anon):    29892 kB
Active(file):    7381376 kB
Inactive(file): 27742324 kB
Unevictable:        1888 kB
Mlocked:            1888 kB
SwapTotal:      33001468 kB
SwapFree:       33001468 kB
Dirty:                80 kB
Writeback:            40 kB
AnonPages:      11769588 kB
Mapped:           457588 kB
Shmem:             31336 kB
KReclaimable:    1266648 kB
Slab:            1819484 kB
SReclaimable:    1266648 kB
SUnreclaim:       552836 kB
KernelStack:       21968 kB
PageTables:        50940 kB
NFS_Unstable:          0 kB
Bounce:                0 kB
WritebackTmp:          0 kB
CommitLimit:    65935788 kB
Committed_AS:   41921324 kB
VmallocTotal:   34359738367 kB
VmallocUsed:      204316 kB
VmallocChunk:          0 kB
Percpu:            48640 kB
HardwareCorrupted:     0 kB
AnonHugePages:  10721280 kB
ShmemHugePages:        0 kB
ShmemPmdMapped:        0 kB
FileHugePages:         0 kB
FilePmdMapped:         0 kB
CmaTotal:              0 kB
CmaFree:               0 kB
HugePages_Total:       0
HugePages_Free:        0
HugePages_Rsvd:        0
HugePages_Surp:        0
Hugepagesize:       2048 kB
Hugetlb:               0 kB
DirectMap4k:      922044 kB
DirectMap2M:    23144448 kB
DirectMap1G:    44040192 kB
```

## CRC version

```bash
crc version: 1.9.0+a68b5e0
OpenShift version: 4.3.10 (embedded in binary)
```
  
## CRC status

(part of the problem)
```bash
ERRO The connection to the server api.crc.testing:6443 was refused - did you specify the right host or port?
 - exit status 1
```

## CRC config

```bash
- cpus                                  : 8
- memory                                : 32768
- pull-secret-file                      : ./pull-secret.txt
```

## Host Operating System

```bash
NAME=Fedora
VERSION="32 (Workstation Edition)"
ID=fedora
VERSION_ID=32
VERSION_CODENAME=""
PLATFORM_ID="platform:f32"
PRETTY_NAME="Fedora 32 (Workstation Edition)"
ANSI_COLOR="0;34"
LOGO=fedora-logo-icon
CPE_NAME="cpe:/o:fedoraproject:fedora:32"
HOME_URL="https://fedoraproject.org/"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora/f32/system-administrators-guide/"
SUPPORT_URL="https://fedoraproject.org/wiki/Communicating_and_getting_help"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=32
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=32
PRIVACY_POLICY_URL="https://fedoraproject.org/wiki/Legal:PrivacyPolicy"
VARIANT="Workstation Edition"
VARIANT_ID=workstation
```

### Steps to reproduce

  1. `crc setup`
  2. `crc config set memory 32768`
  3. `crc config set cpus 8`
  4. `crc config set pull-secret-file ./pull-secret.txt`
  5. `crc start`

### Expected

Being able to login the cluster.

### Actual

`oc login` doesn't work:

```
error: dial tcp 192.168.130.11:6443: connect: connection refused - verify you have provided the correct host and port and that the server is currently running.
```

Neither does `crc status`:

```
ERRO The connection to the server api.crc.testing:6443 was refused - did you specify the right host or port?
 - exit status 1
```

### Logs

You can start crc with `crc start --log-level debug` to collect logs.
Please consider posting this on http://gist.github.com/ and post the link in the issue.

**Comment by cfergeau** [comment]
> You can start crc with crc start --log-level debug to collect logs.
> Please consider posting this on http://gist.github.com/ and post the link in the issue.

Can you paste these logs?

**Comment by ruivieira** [comment]
Sure, they're in here: https://gist.github.com/ruivieira/169dca94a4fd353f14eb3b15f2074d29

Also, `ssh -i ~/.crc/machines/crc/id_rsa coreos@api.crc.testing` returns:

```
coreos@api.crc.testing: Permission denied (publickey,gssapi-keyex,gssapi-with-mic).
```

**Comment by caitelatte** [comment]
Hi folks, I'm having this same set of issues on Fedora 31. It's been pretty repeatable, but has failed in different spots. Most recently I found it was an intermittent issue, as I was able to occasionally sign a CSR (maybe 1 in 20 attempts).

I went through the crc stop, delete, cleanup, start cycle a few times with different levels of verbosity, logs are here: https://gist.github.com/caitelatte/59855abc6d5141768fab0adb2d9823b4

Happy to make a new issue if it's not the same issue after all. :smiley:

**Comment by cfergeau** [comment]
> Also, `ssh -i ~/.crc/machines/crc/id_rsa coreos@api.crc.testing` returns:

Username should be `core`, not `coreos`. Does the api server eventually comes up if you wait a few more minutes after the wait fails? You can check with `nc -v api.crc.testing 6443`

**Comment by cfergeau** [comment]
> I went through the crc stop, delete, cleanup, setup, start cycle a few times with different levels of verbosity, logs are here: https://gist.github.com/caitelatte/59855abc6d5141768fab0adb2d9823b4

How are you generating these logs? Most of the time they contain logs for the preflight checks, and then only the logs from the machine package, which are not the most useful ones. Could you try with `crc start --log-level debug`?

> Happy to make a new issue if it's not the same problem after all. smiley

That would be best as it's hard to say from the start if the issue is going to be the same or not, and helping 2 people at once in the same issue can get confusing for all parties involved.

**Comment by praveenkumar** [comment]
@ruivieira looking at the logs it seems like you tried to run crc on stopped cluster which might not worked at first place https://gist.github.com/ruivieira/169dca94a4fd353f14eb3b15f2074d29#file-crc-start-debug-log-L323-L355 can you try following and share the logs from a fresh start?

```
$ crc delete
$ crc cleanup
$ crc setup
$ crc start --log-level debug
```

**Comment by ruivieira** [comment]
Hi @praveenkumar. [Here are the logs](https://gist.github.com/ruivieira/169dca94a4fd353f14eb3b15f2074d29) after running the above, thanks!

**Comment by praveenkumar** [comment]
@ruivieira Hi, I still think you didn't run the crc from a clean state since as per your logs I don't see `INFO Configuring cluster for first start ` which means either this cluster  is run from a stop state or you run it twice before uploading the logs.

**Comment by kavana-14** [comment]
$  oc login -u developer https://api.crc.testing:6443
error: dial tcp 192.168.130.11:6443: connect: connection refused - verify you have provided the correct host and port and that the server is currently running.

I'm facing the same issue. What is the problem that is not allowing to login to cli or web console?

```
$ crc status
CRC VM:          Running
OpenShift:       Unreachable (v4.15.0-0.okd-2024-02-23-163410)
RAM Usage:       625.2MB of 10.95GB
Disk Usage:      21.48GB of 32.68GB (Inside the CRC VM)
Cache Usage:     27.69GB
Cache Directory: /home/fedora2/.crc/cache

$ crc version
WARN A new version (2.38.0) has been published on https://developers.redhat.com/content-gateway/file/pub/openshift-v4/clients/crc/2.38.0/crc-linux-amd64.tar.xz
CRC version: 2.37.1+36d451
OpenShift version: 4.15.14

```
